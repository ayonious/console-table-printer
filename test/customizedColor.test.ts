import { Table } from '../index';

describe('Example: Print a simple Table with Custom column colors', () => {
  it('Custom column colors are working', () => {
    // Create a table
    const p = new Table({
      columns: [
        { name: 'red_left_align_index', alignment: 'left', color: 'red' },
        { name: 'right_align_text', alignment: 'right' },
        { name: 'green_value', color: 'green' },
      ],
      colorMap: {
        custom_green: '\x1b[32m', // define customized color
      },
    });

    // add rows with custom color
    p.addRow(
      {
        red_left_align_index: 2,
        right_align_text: 'This is my defined Green',
        green_value: 10.212,
      },
      { color: 'custom_green' } // custom color
    );
    p.addRow(
      {
        red_left_align_index: 3,
        right_align_text: 'This row is blue as well',
        green_value: 10.212,
      },
      { color: 'blue' }
    );
    p.addRow(
      {
        red_left_align_index: 4,
        right_align_text: 'This row is green',
        green_value: 10.212,
      },
      { color: 'green' }
    );

    // print
    p.printTable();
    expect(p.render()).toMatchSnapshot();
  });

  it('Disabling color works', () => {
    // Create a table
    const p = new Table({
      columns: [
        { name: 'red_left_align_index', alignment: 'left' },
        { name: 'right_align_text', alignment: 'right' },
        { name: 'green_value' },
      ],
      shouldDisableColors: true,
    });

    // add rows with custom color
    p.addRow(
      {
        red_left_align_index: 2,
        right_align_text: 'This is my defined Green',
        green_value: 10.212,
      },
      { color: 'custom_green' } // custom color
    );
    p.addRow(
      {
        red_left_align_index: 3,
        right_align_text: 'This row is blue as well',
        green_value: 10.212,
      },
      { color: 'blue' }
    );
    p.addRow(
      {
        red_left_align_index: 4,
        right_align_text: 'This row is green',
        green_value: 10.212,
      },
      { color: 'green' }
    );

    // print
    p.printTable();
    expect(p.render()).toMatchSnapshot();
  });
});
