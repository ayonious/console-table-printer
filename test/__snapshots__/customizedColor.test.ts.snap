// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example: Print a simple Table with Custom column colors Custom column colors are working 1`] = `
"┌──────────────────────┬──────────────────────────┬─────────────┐
│[37m [0m[01mred_left_align_index[0m │[37m [0m[01m        right_align_text[0m │[37m [0m[01mgreen_value[0m │
├──────────────────────┼──────────────────────────┼─────────────┤
│[37m [0m[31m2                   [0m │[37m [0m[32mThis is my defined Green[0m │[37m [0m[32m     10.212[0m │
│[37m [0m[31m3                   [0m │[37m [0m[34mThis row is blue as well[0m │[37m [0m[32m     10.212[0m │
│[37m [0m[31m4                   [0m │[37m [0m[32m       This row is green[0m │[37m [0m[32m     10.212[0m │
└──────────────────────┴──────────────────────────┴─────────────┘"
`;
